packages/core/lib/fury.js

```diff
@@ -244,6 +256,33 @@ class Fury {
     return promise;
   }
 
+  /**
+   * Serialize synchronously an API Description into the given output format.
+   *
+   * @param {Object} options
+   * @param {Category} options.api
+   * @param {string} [options.mediaType]
+   */
+  serializeSync({ api, mediaType = 'text/vnd.apiblueprint' }) {
+    const adapter = findAdapter(this.adapters, mediaType, 'serialize');
+
+    if (!adapter) {
+      // eslint-disable-next-line no-console
+      console.warn('Media type did not match any registered serializer!');
+
+      return null;
+    }
+
+    if (!api) {
+      // eslint-disable-next-line no-param-reassign
+      api = new this.minim.elements.Category();
+    }
+
+    return adapter.serialize({
+      api, namespace: this.minim, mediaType, sync: true,
```

Comment:
Some adapters such as `apib-serializer` and `remote` are not respecting this `sync` flag and thus this API would return a promise.

Reply:
Perhaps it might make sense to instead introduce a `serializeSync` method in the adapters. When calling sync version when the adapter doesn't implement that functionality, we could throw an error stating that the particular adapter doesn't support the action as synchronous.

Core could implement the "async" version to use `serializeSync` when `serialize` (async) isn't implemented by an adapter. In the similar way that the `validate` method can be fulfilled by `parse` above.

Reply:
Addressed in 66ab8f2 and fc81a6d